Core/DetectorConstruction.cpp

```diff
@@ -433,7 +476,31 @@ void DetectorConstruction::ConstructLayers(std::vector<G4double>& radius_dynamic
       rot.rotateZ(phi);
       G4ThreeVector loc = G4ThreeVector(radius_new * cos(phi1), radius_new * sin(phi1), 0.0);
       G4Transform3D transform(rot, loc);
-      G4String nameNewI = "scin_" + G4UIcommand::ConvertToString(icopyI + i * 13 + j + 6);
+      moduleNumber = icopyI + i * 13 + j + 6;
+      G4String nameNewI = "scin_" + G4UIcommand::ConvertToString(moduleNumber);
+      
+      if (fCreateGeometryFile) {
+        for (unsigned k=0; k<2; k++) {
+          Channel channTemp(fChannelNumber+k+1, 2*(moduleNumber - 1)+1, k+1, 0);
+          fChannelContainer.push_back(channTemp);
+          
+          Channel channTemp2(fChannelNumber+k+3, 2*(moduleNumber - 1)+2, k+1, 0);
+          fChannelContainer.push_back(channTemp2);
+        }
+        PM pmTemp(2*(moduleNumber - 1)+1, std::to_string(2*(moduleNumber - 1)+1), (2*(moduleNumber - 1)+1)%4 + 1, 
```

Comment:
In PM constructor an argument for "pos_in_matrix" should be set to 0 for PM in Barrel Layers 1-3 and values 1-4 in modular slots (four PM on side A and another 4 on side B)

Reply:
PM structure are deleted, but thank you for pointing that. I corrected that part and moved it to the new file, so maybe in future it will be quicker to add that, if at some point we will be able to simulate lower levels of the analysis like signals, channels and so on.